pkg/component/worker/kubelet.go

```diff
@@ -170,15 +192,25 @@ func (k *Kubelet) Start(ctx context.Context) error {
 		args["--node-labels"] = strings.Join(k.Labels, ",")
 	}
 
-	if runtime.GOOS == "windows" {
-		node, err := node.GetNodename("")
+	nodename, err := node.GetNodename("")
+	if err != nil {
+		return fmt.Errorf("failed to get hostname: %w", err)
+	}
+
+	if k.DualStackEnabled {
+		ipv4, ipv6, err := getIPs(ctx, nodename)
```

Comment:
Can we have a comment with the link to the upstream IP detection for reference? What happens when `--hostname-override` is passed to kubelet?

Reply:
If `--hostname-override` is an ip address, it will be used and no lookups will be made.
 https://github.com/kubernetes/kubernetes/blob/0cc57258c3f8545c8250f0e7a1307fd01b0d283d/pkg/kubelet/nodestatus/setters.go#L200